aicsimageio/readers/reader.py

```diff
@@ -132,8 +157,17 @@ def dask_data(self) -> da.core.Array:
 
     @property
     def data(self) -> np.ndarray:
+        global USE_DASK
         if self._data is None:
             try:
+                print(USE_DASK)
+                # Fast re-route to _read_immediate
+                if not USE_DASK:
```

Comment:
You're rerouting program flow based on a raised exception?

Maybe querying a boolean property might be better. Just seems like a lot of overhead.

Reply:
I agree but if you expand this section below you will see this is the exact section used to route reads so it makes sense to do it this way.